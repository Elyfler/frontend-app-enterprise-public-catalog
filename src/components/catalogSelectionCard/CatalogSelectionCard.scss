.catalog-selection-card{
  flex-direction: row-reverse;
  display: flex;
  justify-content: space-between;
  .pgn__form-checkbox-input{
    margin: 0;
  }
}


