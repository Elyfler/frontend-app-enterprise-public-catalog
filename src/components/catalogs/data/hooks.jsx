const useCourseCatalogs = () => [{ course: 'test', subject: 'test', partner: 'edX' }];

export default useCourseCatalogs;
