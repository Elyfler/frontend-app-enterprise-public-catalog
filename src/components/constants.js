export const paddingClasses = "px-4 px-md-5 px-lg-6"