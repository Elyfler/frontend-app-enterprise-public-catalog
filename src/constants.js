/* eslint-disable import/prefer-default-export */
export const PAGE_TITLE = 'edX Enterprise Catalogs';

export const NUM_RESULTS_PER_PAGE = 40;

/**
 * Event names to use for tracking
 * Spec: https://openedx.atlassian.net/wiki/spaces/AN/pages/2624455676/Enterprise+public+catalog+Event+Design
 */
export const PRODUCT_EVENTS = {
  ctaButtonPressed: 'edx.enterprise.public_catalog.cta_button_pressed',
};
