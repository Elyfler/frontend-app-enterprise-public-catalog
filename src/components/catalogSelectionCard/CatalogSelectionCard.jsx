import React, {
  useContext,
} from 'react';
import PropTypes from 'prop-types';
import { SearchContext, deleteRefinementAction, setRefinementAction } from '@edx/frontend-enterprise';
import {
  Card, Form,
} from '@edx/paragon';
import { QUERY_UUID_REFINEMENT } from '../../constants';

export const CardCheckbox = ({ label, queryUuid }) => {
  const { refinementsFromQueryParams, dispatch } = useContext(SearchContext);
  const isChecked = refinementsFromQueryParams[QUERY_UUID_REFINEMENT]?.includes(queryUuid) || false;
  const setChecked = () => {
    if (isChecked) {
      dispatch(deleteRefinementAction(QUERY_UUID_REFINEMENT));
    } else {
      dispatch(setRefinementAction(QUERY_UUID_REFINEMENT, [queryUuid]));
    }
  };
  return (
    <Form.Checkbox className="catalog-selection-card" checked={isChecked} onChange={setChecked}>{label}</Form.Checkbox>
  );
};

CardCheckbox.propTypes = {
  queryUuid: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

const CatalogSelectionCard = ({ queryUuid, label, cardBody }) => (
  <Card>
    <Card.Body>
      <Card.Title>
        <CardCheckbox label={label} queryUuid={queryUuid} />
      </Card.Title>
      <Card.Text>{cardBody}</Card.Text>
    </Card.Body>
  </Card>
);

CatalogSelectionCard.propTypes = {
  cardBody: PropTypes.string.isRequired,
  queryUuid: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

export default CatalogSelectionCard;
