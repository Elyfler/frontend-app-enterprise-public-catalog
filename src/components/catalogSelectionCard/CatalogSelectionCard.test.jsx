import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { SEARCH_FACET_FILTERS, SearchContext } from '@edx/frontend-enterprise';
import { render, screen } from '@testing-library/react';
import { deleteRefinementAction, setRefinementAction } from '@edx/frontend-enterprise/dist/course-search/data/actions';
import { CardCheckbox } from './CatalogSelectionCard';
import { QUERY_UUID_REFINEMENT } from '../../constants';

const dispatchSpy = jest.fn();
const DEFAULT_SEARCH_CONTEXT_VALUE = { refinementsFromQueryParams: {}, dispatch: dispatchSpy };

// eslint-disable-next-line react/prop-types
const SearchDataWrapper = ({ children, searchContextValue }) => (
  <SearchContext.Provider
    value={searchContextValue}
    searchFacetFilters={
        [...SEARCH_FACET_FILTERS, { attribute: QUERY_UUID_REFINEMENT, title: 'Uuids' }]
      }
  >
    {children}
  </SearchContext.Provider>
);

describe('CardCheckbox', () => {
  beforeEach(() => {
    dispatchSpy.mockReset();
  });
  it('renders a checkbox and label', () => {
    const tree = renderer.create(
      <SearchDataWrapper
        searchContextValue={DEFAULT_SEARCH_CONTEXT_VALUE}
      >
        <CardCheckbox
          label="foo"
          queryUuid="bar"
        />
      </SearchDataWrapper>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('is checked when uuid is included in the query param', () => {
    const queryUuid = 'bar';
    const refinements = { refinementsFromQueryParams: { [QUERY_UUID_REFINEMENT]: [queryUuid] } };
    const tree = renderer.create(
      <SearchDataWrapper
        searchContextValue={refinements}
      >
        <CardCheckbox
          label="foo"
          queryUuid={queryUuid}
        />
      </SearchDataWrapper>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('dispatches setRefinement action when checked', () => {
    const label = 'Check me';
    const queryUuid = 'idforyou';
    render(
      <SearchDataWrapper searchContextValue={DEFAULT_SEARCH_CONTEXT_VALUE}>
        <CardCheckbox
          label={label}
          queryUuid={queryUuid}
        />
      </SearchDataWrapper>,
    );
    const input = screen.getByLabelText(label);
    userEvent.click(input);
    expect(dispatchSpy).toHaveBeenLastCalledWith(setRefinementAction(QUERY_UUID_REFINEMENT, [queryUuid]));
  });
  it('dispatches deleteRefinement action when unchecked', () => {
    const label = 'Uncheck me';
    const queryUuid = 'idforyou';
    const refinements = { refinementsFromQueryParams: { [QUERY_UUID_REFINEMENT]: [queryUuid] }, dispatch: dispatchSpy };
    render(
      <SearchDataWrapper searchContextValue={refinements}>
        <CardCheckbox
          label={label}
          queryUuid={queryUuid}
        />
      </SearchDataWrapper>,
    );
    const input = screen.getByLabelText(label);
    userEvent.click(input);
    expect(dispatchSpy).toHaveBeenLastCalledWith(deleteRefinementAction(QUERY_UUID_REFINEMENT));
  });
});
