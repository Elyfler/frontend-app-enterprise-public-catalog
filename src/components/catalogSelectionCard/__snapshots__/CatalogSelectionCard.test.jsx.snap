// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCheckbox is checked when uuid is included in the query param 1`] = `
<div
  className="pgn__form-checkbox catalog-selection-card"
>
  <input
    checked={true}
    className="pgn__form-checkbox-input"
    id="form-field3"
    onChange={[Function]}
    type="checkbox"
  />
  <div>
    <label
      className="pgn__form-label"
      htmlFor="form-field3"
    >
      foo
    </label>
  </div>
</div>
`;

exports[`CardCheckbox renders a checkbox and label 1`] = `
<div
  className="pgn__form-checkbox catalog-selection-card"
>
  <input
    checked={false}
    className="pgn__form-checkbox-input"
    id="form-field1"
    onChange={[Function]}
    type="checkbox"
  />
  <div>
    <label
      className="pgn__form-label"
      htmlFor="form-field1"
    >
      foo
    </label>
  </div>
</div>
`;
